packages/cms-data-sync/src/utils/assets-and-media.ts

```diff
@@ -0,0 +1,197 @@
+/* eslint-disable no-console */
+
+import {
+  InlineAssetBody,
+  InlineIFrameBody,
+} from 'contentful-html-rich-text-converter';
+import {
+  Asset as ContentfulAsset,
+  Entry,
+  Environment,
+  SysLink,
+} from 'contentful-management';
+import { Asset } from '../autogenerated-gql/graphql';
+import { checkIfEntryAlreadyExistsInContentful } from './entries';
+
+export const checkIfAssetAlreadyExistsInContentful = async (
+  contentfulEnvironment: Environment,
+  id: string,
+) => {
+  try {
+    const asset = await contentfulEnvironment.getAsset(id);
+    return !!asset;
+  } catch (error) {
+    if (error instanceof Error) {
+      const errorParsed = JSON.parse(error?.message);
+      if (errorParsed.status === 404) {
+        return false;
+      }
+    }
+
+    throw error;
+  }
+};
+
+export const processAndPublishEntry = async (
+  entry: ContentfulAsset | Entry,
+) => {
+  try {
+    if ('processForAllLocales' in entry) {
+      const processedEntry = await entry.processForAllLocales();
+      await processedEntry.publish();
+    } else {
+      await entry.publish();
+    }
+  } catch (error) {
+    if (error instanceof Error) {
+      const errorParsed = JSON.parse(error?.message);
+      if (errorParsed.message === 'File has already been processed.') {
+        return;
+      }
+    }
+    throw error;
+  }
+};
+
+export const createAndPublishIfNonExistent = async (
+  type: 'asset' | 'media',
+  contentfulEnvironment: Environment,
+  id: string,
+  checkIfExistsFn: (
```

Comment:
why make this a dependency? any legitimate case for when this isn't the `checkIfAssetAlreadyExistsInContentful` from above?

Reply:
because for the media we need to check if the `entry` exists not the `asset`, so we pass `checkIfEntryAlreadyExistsInContentful` here